flexget/plugins/clients/aria2.py

```diff
@@ -111,16 +115,48 @@ def add_entry(self, aria2, entry, config):
         """
         options = config['options']
         try:
-            options['dir'] = os.path.expanduser(entry.render(config['path']).rstrip('/'))
+            path = entry.get('path', config.get('path', None))
+            options['dir'] = os.path.expanduser(entry.render(path).rstrip('/'))
         except RenderError as e:
             entry.fail('failed to render \'path\': %s' % e)
             return
-        if 'filename' in config:
+
+        filename = entry.get('content_filename', config.get('filename', None))
+        if filename:
+            if config.get('add_extension', False):
+                logger.debug('Getting filename from `{}`', entry['url'])
+                url_info = None
+                try:
+                    ctx = ssl.create_default_context()
+                    ctx.check_hostname = False
+                    ctx.verify_mode = ssl.CERT_NONE
+                    url_info = urlopen(entry['url'], context=ctx)
```

Comment:
Hmm. Using urlopen here, not requests? Is this just to get the headers? A `requests.head` should do this without fetching the full content as well. If we use `task.requests` the user could manually turn off ssl certification with the `verify_ssl` plugin rather than leaving it off permanently here.

Reply:
The problem, is that head returns 404 in several sites, I tested with a lot, and keep getting 404 in some of them! I can only make it work 100% with urlopen, requests.head sometimes fails, even if I make requests with postman with head (from what I read head is buggy in some servers). Is it too bad to check the request with "task.requests.verify" and if False add the "context=ctx"?

Reply:
Maybe if the HEAD request is broken, just do a normal requests.get (or task.requests.get)? I think using requests is preferable to urlopen no matter what.

Reply:
But if I do a request.get I'll download the complete movie/show! No? the idea is just to get the name of the file to be able to make replaces and keep the extension! This is useful to rename url's when no information is provided regarding file name or extension on the url (very common)

Reply:
You can use this method to make sure not to read the body of the response with requests. https://docs.python-requests.org/en/master/user/advanced/#body-content-workflow